syntax/cs.vim

```diff
@@ -44,10 +44,17 @@ syn match	csContextualStatement	/\<partial[[:space:]\n]\+\(class\|struct\|interf
 syn match	csContextualStatement	/\<\(get\|set\)\(;\|[[:space:]\n]*{\)/me=s+3
 syn match	csContextualStatement	/\<where\>[^:]\+:/me=s+5
 
+" Language constructs
+syn keyword	csTypeOf	typeof contained
+syn region	csTypeOfStatement	start="typeof(" end=")" contains=csType, csTypeOf
+syn region	csAttribute	start="^\s*\[" end="\]\s*" contains=csString, csVerbatimString, csCharacter, csNumber, csType, csTypeOfStatement
```

Comment:
I like the idea of an attribute group, but it's tricky to match _all_ attributes while not matching dictionary indexes etc.

```cs
public Function([Attrib] string parameter) {  // This attribute is _not_ highlighted
```

```cs
"this that the other".Split()
    [SomeFunctionThatFindsAnIndex()]; // This array index and the function call it contains _are_ highlighted
var s = someNewtonsoftJsonObject
    ["propertyA"]   // Highlighted as an attribute
    ["propertyD"]   // Highlighted as an attribute
    ["propertyC"];  // Highlighted as an attribute
```

The 2nd examples are a bit contrived but they are valid. All in all I can't really see how to make this work well with vim syntax. What do you think?

Reply:
I didn't know attributes could be applied to function arguments.

I've also realized it only highlights only the first attribute on a line:

        [SerializeField][HideInInspector] string m_Name;

I guess we'd have to have a declaration group that the attribute could be contained within, but that goes beyond my understanding of vim syntax highlighting.

This is something I'd likely keep in my after/syntax/, but will remove from the PR unless I can come up with a more bulletproof solution.